tests/PHPStan/Rules/Functions/ImplodeParameterCastableToStringRuleTest.php

```diff
@@ -0,0 +1,102 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleLevelHelper;
+use PHPStan\Testing\RuleTestCase;
+use const PHP_VERSION_ID;
+
+/**
+ * @extends RuleTestCase<ImplodeParameterCastableToStringRule>
+ */
+class ImplodeParameterCastableToStringRuleTest extends RuleTestCase
+{
+
+	protected function getRule(): Rule
+	{
+		$broker = $this->createReflectionProvider();
+		return new ImplodeParameterCastableToStringRule($broker, new RuleLevelHelper($broker, true, false, true, false, false, true, false));
+	}
+
+	public function testNamedArguments(): void
+	{
+		if (PHP_VERSION_ID < 80000) {
+			$this->markTestSkipped('Test requires PHP 8.0.');
+		}
+
+		$this->analyse([__DIR__ . '/data/param-castable-to-string-functions-named-args.php'], [
```

Comment:
Sorry to be a PITA about this, but each rule should have separate test files. I'd hate to move some lines around and having to modify tests for 3 separate rules...